extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -1568,48 +1583,50 @@ class LifecycleTask extends BackbeatTask {
         if (obj.VersionId) {
             params.VersionId = obj.VersionId;
         }
-        const req = this.s3target.headObject(params);
-        attachReqUids(req, log);
-        return req.send((err, data) => {
-            LifecycleMetrics.onS3Request(log, 'headObject', 'bucket', err);
-
-            if (err) {
+        const command = new HeadObjectCommand(params);
+        attachReqUids(command, log);
+        return this.s3target.send(command)
+            .then(data => {
+                LifecycleMetrics.onS3Request(log, 'headObject', 'bucket', null);
+                const lastModified = data.LastModified instanceof Date 
+                    ? data.LastModified.toISOString() 
+                    : data.LastModified;
```

Comment:
we could do this to cover all cases:
```suggestion
                const lastModified = new Date(data.LastModified).toISOString();
```
...or do you think there is a performance penalty/risk ?

It seems strange we can receive either a Date or "something" else (a string?) : did you see it, or is it some leftover from troubleshooting? Or maybe it handles some null/undefined case, which could & should be handled in a more explicit way?